File: MLSelfSizingCells/MLSelfSizingCells/Views/MLTableViewCell.m
//
//  MLTableViewCell.m
//  MLSelfSizingCells
//
//  Created by Matthias Ludwig on 09.07.15.
//  Copyright © 2015 Matthias Ludwig. All rights reserved.
//

#import "MLTableViewCell.h"

@implementation MLTableViewCell



- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(nullable NSString *)reuseIdentifier {
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        
        self.sentenceLabel = [[UILabel alloc] init];
        self.sentenceLabel.translatesAutoresizingMaskIntoConstraints = NO;
        self.sentenceLabel.numberOfLines = 0;
        self.sentenceLabel.font = [UIFont fontWithName:@"Menlo-Regular" size:25.0f];
        [self.contentView addSubview:self.sentenceLabel];
        
        self.nameLabel = [[UILabel alloc] init];
        self.nameLabel.translatesAutoresizingMaskIntoConstraints = NO;
        self.nameLabel.numberOfLines = 0;
        self.nameLabel.font = [UIFont fontWithName:@"SanFranciscoDisplay-Light" size:12.0f];
        [self.contentView addSubview:self.nameLabel];
        
        self.theImageView = [[UIImageView alloc] init];
        self.theImageView.translatesAutoresizingMaskIntoConstraints = NO;
        self.theImageView.contentMode = UIViewContentModeScaleToFill;
        [self.contentView addSubview:self.theImageView];
        
        NSDictionary *viewsDictionary = @{
                                            @"sentenceLabel": self.sentenceLabel,
                                            @"nameLabel": self.nameLabel,
                                            @"theImageView": self.theImageView
                                          };


        [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|-8-[sentenceLabel]-8-|"
                                                                                        options:kNilOptions
                                                                                        metrics:nil
                                                                                          views:viewsDictionary]];
        
        [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|-20-[sentenceLabel]-10-[nameLabel]"
                                                                                        options: NSLayoutFormatAlignAllLeading | NSLayoutFormatAlignAllTrailing
                                                                                        metrics:nil
                                                                                          views:viewsDictionary]];
        
        [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[nameLabel]-[theImageView(80)]-|" options:kNilOptions metrics:nil views:viewsDictionary]];
        
        [NSLayoutConstraint activateConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"[theImageView(80)]" options:kNilOptions metrics:nil views:viewsDictionary]];
        
        
        [[NSLayoutConstraint constraintWithItem:self.theImageView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self.contentView attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0.0] setActive:YES];
        
        
    }
    
    return self;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: MLSelfSizingCells_Swift/MLSelfSizingCells_Swift/Views/MLTableViewCell.swift
//
//  MLTableViewCell.swift
//  MLSelfSizingCells
//
//  Created by Matthias Ludwig on 25.01.16.
//  Copyright © 2016 Matthias Ludwig. All rights reserved.
//

import UIKit

class MLTableViewCell: UITableViewCell {
    
    let sentenceLabel: UILabel
    let nameLabel: UILabel
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        
        sentenceLabel = UILabel()
        sentenceLabel.translatesAutoresizingMaskIntoConstraints = false
        sentenceLabel.numberOfLines = 0
        sentenceLabel.font = UIFont(name: "Menlo-Regular", size: 25.0)
        
        
        nameLabel = UILabel()
        nameLabel.translatesAutoresizingMaskIntoConstraints = false
        nameLabel.numberOfLines = 0
        nameLabel.font = UIFont(name: "SanFranciscoDisplay-Light", size: 12.0)
        
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        contentView.addSubview(sentenceLabel)
        contentView.addSubview(nameLabel)
        
        let views = [
            "sentenceLabel": sentenceLabel,
            "nameLabel": nameLabel
        ]
        
        NSLayoutConstraint.activateConstraints(NSLayoutConstraint.constraintsWithVisualFormat("|-8-[sentenceLabel]-8-|", options: [], metrics: nil, views: views))
        
        NSLayoutConstraint.activateConstraints(NSLayoutConstraint.constraintsWithVisualFormat("V:|-20-[sentenceLabel]-10-[nameLabel]-20-|", options: [.AlignAllLeading, .AlignAllTrailing], metrics: nil, views: views))
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    
    
}
